textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -42,14 +88,160 @@ public void open() throws Exception {
+		// TODO(Flavio): what if the join predicate has an invalid id attribute?
+		// First check if the field in question exits by using try catch.
+		// Will throw an exception if it doesn't exist. This leads to return false.
+		// Then check if both the fields are of type IntegerField.
+		// --> This is the bare minimum thing that can be done to to verify valid 
+		// id attribute (as of now) (probably it is better to add a field called ID).
+		String fieldName = joinPredicate.getidAttribute().getFieldName();
+		try {
+			if(outerTuple.getField(fieldName).getClass().equals(IntegerField.class)&&
+					innerTuple.getField(fieldName).getClass().equals(IntegerField.class)) {
+				if(outerTuple.getField(fieldName).getValue()==
+						innerTuple.getField(fieldName).getValue()) {
+					return true;
+				}
+			}
+		} catch(Exception e) {
+			;
+		}
+		return false;
+	}
+
+	// Process the tuples to get a tuple with join result if predicate is satisfied.
+	private ITuple processTuples(ITuple outerTuple, ITuple innerTuple, JoinPredicate joinPredicate) throws Exception {
+		ITuple nextTuple = null;
+		List<Span> newJoinSpanList = new ArrayList<>();
+
+		if(!compareId(outerTuple, innerTuple)) {
+			return null;
+		}
+		//TODO(Flavio): getIndex will always return an int, if SchemaConstants.SPAN_LIST doesn't 
+		// exists a NullPointerException will be thrown, it will never return null and the
+		// following comparison will be useless
+		// If either/both tuples have no span information, return null.
+		Integer indexOfInnerSpanList = null;
+		Integer indexOfOuterSpanList = null;
+		try {
+			indexOfInnerSpanList = innerTuple.getSchema().getIndex(SchemaConstants.SPAN_LIST);
+			indexOfOuterSpanList = outerTuple.getSchema().getIndex(SchemaConstants.SPAN_LIST);
+		} catch(Exception e) {
+			return null;
+		}
+		// If either/both tuples have no span information, return null.
+		//		if(indexOfInnerSpanList == null || indexOfOuterSpanList == null) {
+		//			return null;
+		//		}
+
+		List<Span> innerSpanList = null;
+		List<Span> outerSpanList = null;
+		// Check if both the fields obtained from the indexes are indeed of type ListField
+		// TODO(Flavio): should SchemaConstants.SPAN_LIST be a reserved or special 
+		// fieldName? In that case we don't need to check if the classes are equal
+		if(innerTuple.getField(indexOfInnerSpanList).getClass().equals(ListField.class)) {
+			innerSpanList = (List<Span>) innerTuple.getField(indexOfInnerSpanList).getValue();
+		}
+		if(outerTuple.getField(indexOfOuterSpanList).getClass().equals(ListField.class)) {
+			outerSpanList = (List<Span>) outerTuple.getField(indexOfOuterSpanList).getValue();
+		}
+
+		Iterator<Span> outerSpanIter = outerSpanList.iterator();
+
+		while(outerSpanIter.hasNext()) {
+			Span outerSpan = outerSpanIter.next();
+			// Check if the field matches the filed over which we want to join. If not return null.
+			if(!outerSpan.getFieldName().equals(joinPredicate.getjoinAttribute().getFieldName())) {
```

Comment:
I don't think we should just `return null` here.
Different spans in a span list can have different fields. If the field doesn't match for a span, we should proceed to compare next span.